libmscore/paste.cpp

```diff
@@ -324,9 +324,10 @@ bool Score::pasteStaff(XmlReader& e, Segment* dst, int dstStaff)
 
                               // be sure to paste the element in the destination track;
                               // setting track needs to be repeated, as it might have been overwritten by el->read()
-                              el->setTrack(e.track());
+                              // preserve *voice* from source, though
+                              el->setTrack(e.track() + el->voice());
```

Comment:
I don't get all the ins and outs just yet but it feels definitely better and more understandable to use dstStaffIdx * VOICES + voice instead of relying to track(). Even if it's set at the start of the loop to dstStaffIdx * VOICES